broadinstitute/single_cell_portal_core
# Adds auth error messages to scp-api.js calls

Previously, authentication errors were not handled in a uniform way, and many resulted in unhelpful errors displayed to users. This change pops a single, standard modal any time an ajax call from scp-api returns a 401/403

![image](https://user-images.githubusercontent.com/2800795/121760725-3bde6f00-cafa-11eb-9f23-c86b1c1a8717.png)

This modal dedupes these errors, so only a single modal & message will be displayed if there are multiple simultaneous errors (which is common for the explore tab, as multiple plots will all have requests that fail). 

TO TEST:
 1.  edit the first line of the `show` method in  `clusters_controller.rb` to be  `render json: {error: 'Unauthorized'}, status: 403 and return`
  2. load a study 
  3.  confirm a modal pops up as above
